data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/contingency/CsaProfileContingencyCreator.java

```diff
@@ -57,53 +60,63 @@ public CsaProfileContingencyCreator(Crac crac, Network network, PropertyBags con
+        csaProfileContingencyCreationContexts = new HashSet<>();
 
         for (PropertyBag contingencyPropertyBag : contingenciesPropertyBags) {
-            this.addContingency(contingencyPropertyBag);
+            String contingencyId = contingencyPropertyBag.getId(REQUEST_CONTINGENCY);
+            try {
+                addContingency(contingencyPropertyBag);
+            } catch (OpenRaoImportException exception) {
+                csaProfileContingencyCreationContexts.add(CsaProfileElementaryCreationContext.notImported(contingencyId, exception.getImportStatus(), exception.getMessage()));
+            }
         }
-        this.cracCreationContext.setContingencyCreationContexts(csaProfileContingencyCreationContexts);
+
+        cracCreationContext.setContingencyCreationContexts(csaProfileContingencyCreationContexts);
     }
 
     private void addContingency(PropertyBag contingencyPropertyBag) {
 
         String contingencyId = contingencyPropertyBag.getId(CsaProfileConstants.REQUEST_CONTINGENCY);
-        Set<PropertyBag> contingencyEquipments = this.dataCheck(contingencyPropertyBag, contingencyId);
-        if (contingencyEquipments.isEmpty()) {
-            return;
+        String contingencyName = CsaProfileCracUtils.createElementName(contingencyPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_NAME), contingencyPropertyBag.getId(CsaProfileConstants.REQUEST_CONTINGENCIES_EQUIPMENT_OPERATOR)).orElse(contingencyId);
+        List<String> alterations = new ArrayList<>();
+
+        checkMustStudy(contingencyId, Boolean.parseBoolean(contingencyPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_NORMAL_MUST_STUDY)));
+        Set<PropertyBag> contingencyEquipments = contingencyEquipmentsPropertyBags.get(contingencyId);
+
+        ContingencyAdder contingencyAdder = crac.newContingency().withId(contingencyId).withName(contingencyName);
+        addContingencyEquipments(contingencyId, contingencyEquipments, contingencyAdder, alterations);
+        contingencyAdder.add();
+
+        csaProfileContingencyCreationContexts.add(CsaProfileElementaryCreationContext.imported(contingencyId, contingencyId, contingencyName, String.join(" ", alterations), !alterations.isEmpty()));
+    }
+
+    private static void checkMustStudy(String contingencyId, Boolean mustStudy) {
+        if (!Boolean.TRUE.equals(mustStudy)) {
+            throw new OpenRaoImportException(ImportStatus.NOT_FOR_RAO, formatNotImportedMessage(contingencyId, "its field mustStudy is set to false"));
+        }
+    }
+
+    private void addContingencyEquipments(String contingencyId, Set<PropertyBag> contingencyEquipments, ContingencyAdder contingencyAdder, List<String> alterations) {
+        if (contingencyEquipments == null || contingencyEquipments.isEmpty()) {
+            throw new OpenRaoImportException(ImportStatus.INCOMPLETE_DATA, formatNotImportedMessage(contingencyId, "no contingency equipment is linked to the contingency"));
```

Comment:
to be more precise when combined to the formatting prefix , use:  linked to 'that' contingency ,